src/Data/Int.purs

```diff
@@ -77,6 +79,21 @@ foreign import toNumber :: Int -> Number
 fromString :: String -> Maybe Int
 fromString = fromStringAs (Radix 10)
 
+-- | A type for describing whether an integer is even or odd.
+data Parity = Even | Odd
+
+derive instance eqParity :: Eq Parity
+derive instance ordParity :: Ord Parity
```

Comment:
I'm thinking of this as being an approximation to integers, not Z/2Z. Dividing an even integer by another even integer could be even or odd.